integration-tests/webhook/webhook_test.go

```diff
@@ -0,0 +1,163 @@
+package webhook
+
+import (
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openebs/maya/integration-tests/artifacts"
+	k8s "github.com/openebs/maya/pkg/client/k8s/v1alpha1"
+	cvr "github.com/openebs/maya/pkg/cstorvolumereplica/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// stsYaml holds the yaml spec
+	// for statefulset application
+	clonePVCYaml artifacts.Artifact = `
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: demo-snap-vol-claim
+  annotations:
+    snapshot.alpha.kubernetes.io/snapshot: fastfurious
+spec:
+  storageClassName: openebs-snapshot-promoter
+  accessModes: [ "ReadWriteOnce" ]
+  resources:
+    requests:
+      storage: 2G`
+)
+
+var _ = Describe("AdmissionWebhook", func() {
```

Comment:
Let us explain what we are trying to do in describe; along with tags.

Reply:
We already have individual level `context` and `Description` as well as `By`  for each state.